### DISCUSSION TITLE
frBlockObjectEnum in TritonRoute

### DISCUSSION THREAD

**Discussion by gilgamsh** [The-OpenROAD-Project/OpenROAD#3252] [Q&A]
I am learning the code of TritonRoute, and I am a little confused about the `route_queue_main()` in `FlexDR_maze.cpp`, In the `rerouteQueue` the entry maybe `drcNet,frcNet,frcInst,frcInstBlockage`, what is the meaning of these enum? Thank you very much.

**Comment by maliberty** [comment]
If you have a frBlockObject* the enum would tell you which of the derived classes it is pointing to.  In what scenario would a frcInst or frcInstBlockage appear in the queue?
